### PR TITLE
[Feature] Quantization global optimization

### PR DESCRIPTION
**Pull request by humu789**
43563e8e-5b15-466f-a341-a3e2b8e87667
ptq quantized and deployed experiment results on openvino and trt backends

mainly changes:
1. add mmcls and mmdet models ptq configs on tensorrt
3. change ptq calibration dataset from train to val
4. unify calibration batch size and calibration steps
5. replace fakequant type and freeze params before deployment
6. optimaze handle pipeline for w_fake_quant before deployment
7. optimaze tensorrt backendconfig
8. optimaze interfaces of quantizers

### CODE REVIEW

File: mmrazor/models/algorithms/quantization/mm_architecture.py [hunk 1]
```diff
@@ -348,17 +349,25 @@ def get_deploy_model(self):
...
-
         observed_model = self.quantizer.prepare(fp32_model)
-
         observed_model.load_state_dict(quantized_state_dict)
 
         self.quantizer.post_process_for_deploy(
-            observed_model, keep_w_fake_quant=True)
+            observed_model, device=device, keep_w_fake_quant=True)
+
+        for node in observed_model.graph.nodes:
```
Comment (HIT-cwh) [inline mmrazor/models/algorithms/quantization/mm_architecture.py:362]: 最好这里加行注释解释下

File: mmrazor/structures/quantization/qconfig.py [hunk 2]
```diff
@@ -5,10 +5,13 @@
...
 
 try:
-    from torch.ao.quantization import QConfig
+    from torch.ao.quantization import FakeQuantize, QConfig
 except ImportError:
     from mmrazor.utils import get_placeholder
     QConfig = get_placeholder('torch>=1.13')
+    FakeQuantize = get_placeholder('torch>=1.13')
+
+from torch.ao.quantization.utils import is_per_tensor
```
Comment (HIT-cwh) [inline mmrazor/structures/quantization/qconfig.py:14]: 这里可能需要用下get_placeholder支持torch<1.13，但我之前在我的pr里改了，这里先合了应该也可以

File: mmrazor/models/quantizers/native_quantizer.py [hunk 3]
```diff
@@ -275,36 +279,46 @@ def traverse(module):
...
+                            float_child[0].weight.data = weight_fakequant(
+                                float_child[0].weight.data)
+                        else:
+                            float_child.weight.data = weight_fakequant(
+                                float_child.weight.data)
                     # This is decided by backend type, some backend need
                     # explicitly keep the fake quant structure, others don't.
                     # TODO add deploy doc link
                     if keep_w_fake_quant:
+                        self.qconfig.fixed_w_fakequant()
```
Comment (HIT-cwh) [inline mmrazor/models/quantizers/native_quantizer.py:300]: 感觉最好加一行注释解释下为啥需要fix？

**Comment by HIT-cwh** [comment]
1. release下对过精度的onnx文件？

File: mmrazor/models/algorithms/quantization/mm_architecture.py (hunk 1 above)
Comment (humu789) [reply mmrazor/models/algorithms/quantization/mm_architecture.py:362]: 这里不仅仅是为了处理lsq的，各种fakequant都替换成基本的fakequant，方便后续统一转换

**Review by pppppM — APPROVED** [review]

**Pull request merged**